src/picongpu/include/fields/FieldJ.kernel

```diff
@@ -361,83 +361,145 @@ struct KernelAddCurrentToEMF
     }
 };
 
+template< uint32_t T_workers >
 struct KernelBashCurrent
 {
-    template<class Mapping>
-    DINLINE void operator()(J_DataBox fieldJ,
-                                      J_DataBox targetJ,
-                                      DataSpace<simDim> exchangeSize,
-                                      DataSpace<simDim> direction,
-                                      Mapping mapper) const
+    template< typename Mapping >
+    DINLINE void operator()(
+        J_DataBox const & fieldJ,
```

Comment:
I hoped something like this but as I had no idea about the implementation of `PMACC_KERNEL`, I thought it is better to ask ;)